assayist/processor/utils.py

```diff
@@ -146,14 +204,30 @@ def unpack_artifacts(artifacts, output_dir):
         if not os.path.isfile(artifact):
             raise RuntimeError(f'The artifact "{artifact}" could not be found')
 
-        log.info(f'Unpacking {os.path.split(artifact)[-1]}')
-        # Create a subdirectory to store the unpacked artifact
-        output_subdir = os.path.join(output_dir, os.path.split(artifact)[-1])
-        if not os.path.isdir(output_subdir):
-            os.mkdir(output_subdir)
+        artifact_filename = os.path.split(artifact)[-1]
+        log.info(f'Unpacking {artifact_filename}')
+
+        if artifact_filename.startswith('docker-image') and artifact_filename.endswith('.tar.gz'):
+            output_subdir = os.path.join(output_dir, 'container_layer', artifact_filename)
+            os.makedirs(output_subdir)
+            unpack_container_image(artifact, output_subdir)
 
-        extension = os.path.splitext(artifact)[-1]
-        if extension == '.rpm':
+        elif artifact_filename.endswith('.rpm'):
+            output_subdir = os.path.join(output_dir, 'rpm', artifact_filename)
+            os.makedirs(output_subdir)
             unpack_rpm(artifact, output_subdir)
+
+        elif zipfile.is_zipfile(artifact):
+            output_subdir = os.path.join(output_dir, 'non-rpm', artifact_filename)
+            os.makedirs(output_subdir)
+            unpack_zip(artifact, output_subdir)
+
+        elif tarfile.is_tarfile(artifact):
+            output_subdir = os.path.join(output_dir, 'non-rpm', artifact_filename)
+            os.makedirs(output_subdir)
+            unpack_tar(artifact, output_subdir)
+
         else:
-            raise RuntimeError(f'"{artifact}" is not a supported file type to unpack')
+            # Files such as .pom don't to be unpacked, others such as .gem are not yet supported.
+            log.info(f'Skipping unpacking (unsupported archive type or not an archive): {artifact}')
```

Comment:
Should we have a different log message based on if it's one that doesn't need unpacking vs an archive we don't support?

Reply:
That would require us to store a list of extensions that don't require unpacking or those that we don't support yet. There is quite a few:

```
       name       |                       extensions                       
------------------+--------------------------------------------------------
 sig              | sig signature
 cab              | cab
 zip              | zip
 txt              | txt
 ini              | ini
 wsf              | wsf
 manifest         | manifest
 spec             | spec
 exe              | exe
 dll              | dll
 lib              | lib
 sys              | sys
 inf              | inf
 cat              | cat
 msi              | msi
 pdb              | pdb
 oem              | oem
 gem              | gem
 pom              | pom
 ova              | ova
 xml              | xml
 jar              | jar war rar ear sar kar jdocbook jdocbook-style plugin
 js               | js
 iso              | iso
 so               | so
 raw              | raw
 qcow             | qcow
 qcow2            | qcow2
 ldif             | ldif
 cfg              | cfg
 xsd              | xsd
 properties       | properties
 tar              | tar tar.gz tar.bz2 tar.xz
 msm              | msm
 vdi              | vdi
 aar              | aar
 vmdk             | vmdk
 ks               | ks
 apklib           | apklib
 dylib            | dylib
 vhd              | vhd
 box              | box
 groovy           | groovy gvy
 pear             | pear
 typescript       | d.ts
 json             | json
 key              | key
 dot              | dot gv
 batch            | bat
 shell            | sh
 rc               | rc
 wsdl             | wsdl
 obr              | obr
 py               | py
 html             | html htm
 xslt             | xslt
 tlb              | tlb
 liveimg-squashfs | liveimg.squashfs
 target           | target
 jnilib           | jnilib
 raw-xz           | raw.xz
 yaml             | yaml yml
 xjb              | xjb
 xmlcompressed    | xml.gz xml.bz2 xml.xz
(64 rows)
```

I'm not sure this is worth it. Wdyt @mprahl?

Reply:
Probably not. Let's skip for now.